.github/workflows/requirements-check.yml

```diff
@@ -6,12 +6,19 @@ jobs:
   requirements-check:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - name: Install uv
-      uses: astral-sh/setup-uv@v6
-      with:
-        version: "0.7.x"
-    - name: Install dependencies
-      run: uv sync --frozen
-    - run: uv run ./requirements/update_requirements.py
-    - run: git diff --exit-code
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.x"
+          cache: pip
+
+      - name: Install uv
+        run: pip install --upgrade 'uv~=0.7.0'
```

Comment:
I've used zizmor to check the security in #2, and while it enforces the usage of a commit hash, it doesn't disallow the usage of third party actions.
`setup_uv` also deals with caching, which could be nice given large enough dependencies

Reply:
I've removed this commit